src/main/scala/uk/gov/nationalarchives/omega/api/services/ApiService.scala

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2023 The National Archives
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.gov.nationalarchives.omega.api.services
+
+import cats.effect.std.Queue
+import cats.effect.{ ExitCode, IO }
+import cats.implicits.catsSyntaxParallelTraverse_
+import jms4s.JmsAcknowledgerConsumer.AckAction
+import jms4s.config.QueueName
+import jms4s.jms.JmsMessage
+import jms4s.{ JmsAcknowledgerConsumer, JmsProducer }
+import org.typelevel.log4cats.slf4j.Slf4jFactory
+import org.typelevel.log4cats.{ LoggerFactory, SelfAwareStructuredLogger }
+import pureconfig.generic.auto._
+import pureconfig.{ ConfigObjectSource, ConfigSource }
+import uk.gov.nationalarchives.omega.api.common.ServiceArgs
+import uk.gov.nationalarchives.omega.api.conf.ServiceConfig
+import uk.gov.nationalarchives.omega.api.connectors.JmsConnector
+import uk.gov.nationalarchives.omega.api.services.LocalMessageStore.PersistentMessageId
+import uk.gov.nationalarchives.omega.api.services.ServiceState.{ Started, Starting, Stopped, Stopping }
+
+import java.nio.file.{ Files, Paths }
+import java.util.concurrent.atomic.AtomicReference
+
+class ApiService(val config: ServiceConfig) {
+
+  implicit val loggerFactory: LoggerFactory[IO] = Slf4jFactory[IO]
+  implicit val logger: SelfAwareStructuredLogger[IO] = LoggerFactory[IO].getLogger
+
+  private var errorOnInvalidServiceState = false
+  private val state = new AtomicReference[ServiceState](Stopped)
+
+  private val localMessageStore = {
+    val messageStoreFolder = Paths.get(config.tempMessageDir)
```

Comment:
You could use the `Files.createTempDirectory` to make it portable.

Reply:
That won't work because the point is to preserve the messages in the event of a crash. If the system reboots then files in the temp directory will be lost.

Reply:
Only a suggestion: resolving.